tests/testthat/test-new_name.R

```diff
@@ -6,7 +6,6 @@ test_that("takes a custom `board`", {
 
 test_that("uses a default board", {
   default <- board_reuse()
-  withr::defer(pins::pin_delete(default, "old"))
```

Comment:
This is important to keep because here we're not using a self-destroying temporary board. Instead we're using a permanent board in the host computer, and we want to avoid leaving behind artifacts of our tests. Worse, with "old" left behind in the default board the following tests may behave in an unexpected way.

You may see `?on.exit()` or `?withr::defer()`, or better https://testthat.r-lib.org/articles/test-fixtures.html

Reply:
This expression actually made me realise that we had a small mistake in the `new_name` function: 

- Originally:  `pins::pin_delete(board, old)`
- Should be:  pins::pin_delete(board, "old"), right?

Reply:
I think I understand what this expression is supposed to do but I don't understand why it makes the test fail in this way, especially when test passes without this expression: 
```
Error (???): uses a default board
<pins_pin_missing/rlang_error/error/condition>
Error in `abort_pin_missing(name)`: Can't find pin called 'old'
```

Reply:
Thanks Julie. The bug was in my call to `defer()`: After calling `new_name()` the pin "old" doesn't exist because we delete it in `new_name()` to the error message in the test make sense. If anything needs to be cleaned up after the test is the pin "new". So I fixed the call to `defer()` to remove not "old" but "new". You'll see my commit.

BTW,  I think `new_name()` is correct because inside its the body the object `old` holds whatever literal string the user passed through the argument `old`.

The situation seems similar to this one:

``` r
f <- function(old) {
  old
}

f("banana")
#> [1] "banana"
# Same
old <- "banana"
f(old)
#> [1] "banana"
```